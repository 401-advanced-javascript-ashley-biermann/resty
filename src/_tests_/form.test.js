import React from 'react';

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Form from '../components/form/form.js';

describe('Form tests', () => {
  it('Should render at application start ', () => {
    let app = shallow(<Form />);
    expect(app.find('#form').exists()).toBe(true);
    expect(app.find('form').exists()).toBe(true);
    expect(app.find('#methods').exists()).toBe(true);
    expect(app.find('#apiRequest').exists()).toBe(true);
  });
  it('Should update method on click of radio buttons', () => {
    let app = mount(<Form />);
    // let getButton = 
  });
});