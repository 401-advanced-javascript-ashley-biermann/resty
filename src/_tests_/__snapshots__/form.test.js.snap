// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Form component renders correctly 1`] = `
<div
  id="form"
>
  <form
    onSubmit={[Function]}
  >
    <legend>
      Enter API URL, and click on desired METHOD
    </legend>
    <input
      id="urlInput"
      onChange={[Function]}
      placeholder="URL"
      type="text"
      value=""
    />
    <input
      className="button"
      id="submitButton"
      type="submit"
      value="GO"
    />
  </form>
  <section
    id="methods"
  >
    <input
      className="button"
      id="get"
      onClick={[Function]}
      type="button"
      value="GET"
    />
    <input
      className="button"
      id="post"
      onClick={[Function]}
      type="button"
      value="POST"
    />
    <input
      className="button"
      id="put"
      onClick={[Function]}
      type="button"
      value="PUT"
    />
    <input
      className="button"
      id="patch"
      onClick={[Function]}
      type="button"
      value="PATCH"
    />
    <input
      className="button"
      id="delete"
      onClick={[Function]}
      type="button"
      value="DELETE"
    />
  </section>
  <section
    id="apiRequest"
  >
    <p>
      
    </p>
  </section>
</div>
`;
